src/Operations/ConvertContoursToMeshes.cc

```diff
@@ -264,6 +264,36 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
             return false;
         };
 
+        const auto projected_contours_intersect = [&](const plane<double> &pln_A, cop_refw_t A,
+                                                            const plane<double> &pln_B, cop_refw_t B) -> bool {
+            // Checks whether two contors intersect
+            // if a contour is completely enclosed by another contour, they will not intersect
+            // Given that the contours overlap, if one does not enclose the other, they must intersect.
+            // A countour is enclosed if *all* vertices of a contour lies inside the other polygon.
+            // Should work well for most realistic contours that are highly sampled
+
+            if (!projected_contours_overlap(pln_A, A, pln_B, B)){
+                return false;
+            }
+
+            bool a_in_b = true;
+            bool b_in_a = true;
+
+            for(const auto &p_A : A.get().points){
+                if(!B.get().Is_Point_In_Polygon_Projected_Orthogonally(pln_B, p_A)){
+                    a_in_b = false;
```

Comment:
can't you return false immediately?

Reply:
No, because if a is not enclosed by b, you still need to check whether b is enclosed by a before you can tell weather the contours intersect or not. But I think I can change the logic around to make things a bit more efficient (albeit slightly less readable) so I could commit that. It would look like this:

```
for(const auto &p_A : A.get().points){
     if(!B.get().Is_Point_In_Polygon_Projected_Orthogonally(pln_B, p_A)){
         for(const auto &p_B : B.get().points){
             if(!A.get().Is_Point_In_Polygon_Projected_Orthogonally(pln_A, p_B)){
                 return true
             }
         }
     }
}
return false
```

Is this preferable?

Reply:
If a is not enclosed by b, doesn't that mean it automatically intersects?

Not sure if we want nested for loops, you can also just add a break condition after = false for both loops as you have it currently

Reply:
If a is not enclosed by b, it is still possible that b could be enclosed by a, so it doesn't automatically mean that the contours intersect.

I agree about the break condition though, I'll add that